### PR TITLE
Adding Github Actions for podman testing

### LINKED ISSUE #6340: Setup CI infra for testing odo on Podman
@## User Story
As an odo developer, I want to set up CI infra So that I can reliably run `odo` tests using Podman as backend and be confident with my code changes.

## Acceptance Criteria
- [ ] It should not keep a long-running VM; instead, it should spin up VMs (with Podman) on the fly and destroy them once the tests are done
- [ ] It should pass all tests added in https://github.com/redhat-developer/odo/pull/6332, using the `test-integration-podman` Make target
- [ ] Add a script to reproduce the env.

## Links 
- Related Epic (mandatory): [#6332 ](https://github.com/redhat-developer/odo/issues/5871)
- Parent user story: https://github.com/redhat-developer/odo/issues/6236

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this:**

/kind tests

**What does this PR do / why we need it:**
This PR Adds GitHub Actions to tests odo against podman.

**Which issue(s) this PR fixes:**

Fixes #6340

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: scripts/ansible/VM/create.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+---
```
Comment (feloy) [inline scripts/ansible/VM/create.yml:1]: All resources are installed in the Default Resource Group. Please create a specific Resource Group (with name based on `name_prefix`) for these resources, and put all created resources into it

File: scripts/ansible/VM/create.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    - name: Checking for existing VPC
+      ibm_is_vpc_info:
+        name: "{{ vpc_name }}"
+      failed_when:
+        - vpc.rc != 0
+        - '"No VPC found" not in vpc.stderr'
+      register: vpc
+
+    - name: Configure VPC
```
Comment (feloy) [inline scripts/ansible/VM/create.yml:19]: The script is not idempotent, as it should be. When I run it again, I get an error at the step: TASK [Configure VPC]

File: scripts/ansible/VM/create.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+# above this line all the steps are working 
+# next steps are for installation of podman and other similar configurations
+
+- name: copy installation scirpt
+  hosts: new_vsi
+  tasks:
+    - name: copying file with playbook
+      become: true 
+      copy:
+        src: ~/go/src/github.com/redhat-developer/odo/scripts/ansible/VM/install_dependency.sh
```
Comment (feloy) [inline scripts/ansible/VM/create.yml:155]: Please do not use absolute paths

File: scripts/ansible/VM/create.yml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        dest: ~/      
+        mode: 0644
+    - name: install podman and golang
+      command: bash ~/install_dependency.sh
+      register: output
```
Comment (feloy) [inline scripts/ansible/VM/create.yml:160]: I would prefer to use Ansible directives to install necessary packages, instead of a script. The Ansible directives are generally compatible with different Linux distributions, which can make life much easier whzn we need to change distribution

**Review by feloy — CHANGES_REQUESTED** [review]

File: .github/workflows/podman-test.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+name: GitHub Actions Demo
```
Comment (feloy) [inline .github/workflows/podman-test.yaml:1]: Could you please change the name of the Action, so it is easily findable in the list?

**Review by feloy — CHANGES_REQUESTED** [review]
I would be happy to merge the GitHub action. Could you remove the other files that are not necessary for the moment?

File: .github/workflows/podman-test.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+on:
+  pull_request:
+    branches:    
+      - main
+
+jobs:
+  Explore-GitHub-Actions:
```
Comment (rm3l) [inline .github/workflows/podman-test.yaml:9]: Can you also change the job name to reflect what it really does?

File: .github/workflows/podman-test.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '^1.18'
```
Comment (rm3l) [inline .github/workflows/podman-test.yaml:19]: To avoid duplication, I think the `setup-go` Action is able to pick the Go version specified in `go.mod`. See https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

File: .github/workflows/podman-test.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+    - name: Build odo
+      run: make install
+
+    - name: Run Integration tests
+      run: |
+        make test-integration-podman
```
Comment (rm3l) [inline .github/workflows/podman-test.yaml:26]: Because we are also generating JUnit reports, I think it would be nice also to upload/display such reports, as it would make analysis much easier.
So I was thinking of a final step to either [upload](https://github.com/actions/upload-artifact) the JUnit report as an artifact or, better, parse and display it (e.g., via [`actions/test-reporter`](https://github.com/marketplace/actions/test-reporter))? This new step would run regardless of the status of the "Run Integration tests" step.
(But no problem, this can be done later on in a separate issue/PR)

**Review by rm3l — CHANGES_REQUESTED** [review]

**Comment by anandrkskd** [comment]
We can use create edge case that we discussed in [here](https://github.com/redhat-developer/odo/issues/6298#issuecomment-1342672316), by using config for podman, [here](https://github.com/redhat-developer/odo/actions/runs/3829115665/jobs/6515451294#step:3:7) is a test run that checked that part.

@feloy, @rm3l, do you want me to remove the config file for podman as well, as it could be used in tests in future?

**Comment by feloy** [comment]
@anandrkskd during the test, I can see you get this error:

```

**Comment by rm3l** [comment]
I can see the `test-reporter` action is failing because of `Error: TypeError: stackTrace.split is not a function`..
To not block this PR, it is okay to me if we implement the reporting part in a separate PR if you prefer..

**Comment by rm3l** [comment]
@anandrkskd Can you also update the title of this PR to reflect what this PR now does?
The title will show up as the squash commit message in the Git history, and as it is it does not help to understand what this commit does.

File: .github/workflows/podman-test.yaml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: 'go.mod'
+
+    - name: Build odo
+      run: make install
+
+    - name: Run Integration tests
```
Comment (rm3l) [inline .github/workflows/podman-test.yaml:23]: As discussed yesterday, can you explicitly prevent the job from failing when this step does not pass? This is temporary until we check the compatibility with podman v3 (https://github.com/redhat-developer/odo/issues/6446).

```suggestion
    - name: Run Integration tests
      # TODO Make the job fail when this step fails once we check the compatibility with podman v3.
      # See https://github.com/redhat-developer/odo/issues/6446
      continue-on-error: true
```

**Review by rm3l — APPROVED** [review]

**Comment by rm3l** [comment]
```
• [FAILED] [45.875 seconds]
odo logs command tests
/go/odo_1/tests/integration/cmd_logs_test.go:14
  when component is created and odo logs is executed
  /go/odo_1/tests/integration/cmd_logs_test.go:69
    when running in Deploy mode
    /go/odo_1/tests/integration/cmd_logs_test.go:129
      [It] should successfully show logs of the running component
      /go/odo_1/tests/integration/cmd_logs_test.go:136

...
Expected
      <string>: main: Thu Jan  5 13:14:44 UTC 2023 - this is infinite while loop
      main[1]: 
      main[1]: > nodejs-starter@1.0.0 start /opt/app-root/src
      main[1]: > node server.js
      main[1]: 
      main[1]: App started on PORT 3000
      
  to contain substring
      <string>: main[2]:
  In [It] at: /go/odo_1/tests/helper/helper_generic.go:79
...
Summarizing 1 Failure:
  [FAIL] odo logs command tests when component is created and odo logs is executed when running in Deploy mode [It] should successfully show logs of the running component
  /go/odo_1/tests/helper/helper_generic.go:79

Ran 343 of 603 Specs in 1155.129 seconds
FAIL! -- 342 Passed | 1 Failed | 0 Pending | 260 Skipped

```

Overriding because this is not related to this PR, but I'm taking a look at this issue as I saw this error reported on more than 2 other PRs.

/override OpenShift-Integration-tests/OpenShift-Integration-tests

**Comment by rm3l** [comment]
Infra errors on Prow.

/override ci/prow/v4.11-integration-e2e
